Source/Scene/Scene.js

```diff
@@ -3038,13 +3066,7 @@ define([
         this._postUpdate.raiseEvent(this, time);
 
         var cameraChanged = checkForCameraUpdates(this);
-        var shouldRender = !this.requestRenderMode || this._renderRequested || cameraChanged || (this.mode === SceneMode.MORPHING);
-        if (!shouldRender && defined(this.maximumRenderTimeChange) && defined(this._lastRenderTime)) {
-            var difference = Math.abs(JulianDate.secondsDifference(this._lastRenderTime, time));
-            shouldRender = shouldRender || difference >= this.maximumRenderTimeChange;
-        }
-
-        if (shouldRender) {
+        if (shouldRender || cameraChanged) {
```

Comment:
Don't you want to call your `shouldRender` function somewhere in here?  Otherwise this is always true